solidity-mode.el

```diff
@@ -541,6 +541,7 @@ Cursor must be at the function's name.  Does not currently work for constructors
   (set (make-local-variable 'comment-multi-line) t)
   (set (make-local-variable 'comment-line-break-function)
        'c-indent-new-comment-line)
+  (set (make-local-variable 'c-mode-hook) nil)
```

Comment:
All right this seems like a kind of okayish work-around for solidity mode only unless someone wants to spend the time to make the mode derive from prog-mode and not c-mode.

Can you also add a comment above explaining why this line is there and also linking to issue https://github.com/ethereum/emacs-solidity/issues/49?

Reply:
ok.

I found PHP mode is also derived from c-mode.(https://github.com/emacs-php/php-mode/blob/d3a61b19f3fe0413740eb94efd9fdd53e1157153/php-mode.el#L1143)

So I think this work-around is not very hacky.

Reply:
Nice, I also had seen someother modes derive c-mode. Can you maybe then do the following also? Create a configuration variable like this one: https://github.com/emacs-php/php-mode/blob/d3a61b19f3fe0413740eb94efd9fdd53e1157153/php-mode.el#L339-L343

and call it `solidity-mode-disable-c-mode-hook`

And then set the hook to nil only if that is `t` as it's done here: https://github.com/emacs-php/php-mode/blob/d3a61b19f3fe0413740eb94efd9fdd53e1157153/php-mode.el#L1150-L1151